Add new chart: Keycloak (#4163)
bitnami/charts: bitnami/keycloak/README.md

| `image.pullPolicy`                      | Keycloak image pull policy                                                               | `IfNotPresent`                                          |
| `image.pullSecrets`                     | Specify docker-registry secret names as an array                                         | `[]` (does not add image pull secrets to deployed pods) |
| `image.debug`                           | Specify if debug logs should be enabled                                                  | `false`                                                 |
| `auth.createAdminUser`                  | Create administrator user on boot                                                        | `true`                                                  |
| `auth.adminUser`                        | Keycloak administrator name                                                              | `user`                                                  |
| `auth.adminPassword`                    | Keycloak administrator password for the new user                                         | _random 10 character long alphanumeric string_          |

> | `auth.managementUser`                   | Wildfly management user                                                                  | `user`                                                  |

Reviewer: ```suggestion
| `auth.managementUser`                   | Wildfly management user                                                                  | `manager`                                               |
```

Small typo.

| `auth.tls.jksSecret`                    | Existing secret containing the truststore and one keystore per Keycloak replica          | `nil`                                                   |
| `auth.tls.keystorePassword`             | Password to access the keystore when it's password-protected                             | `nil`                                                   |
| `auth.tls.truststorePassword`           | Password to access the truststore when it's password-protected                           | `nil`                                                   |
| `bindAddress`                           | Keycloak bind address                                                                    | `0.0.0.0`                                               |
| `proxyAddressForwarding`                | Enable Proxy Address Forwarding                                                          | `false`                                                 |
| `serviceDiscovery.enabled`              | Enable Service Discovery for Keycloak (required if `replicaCount` > `1`)                 | `false`                                                 |

> | `serviceDiscovery.protocol`             | Sets the protocol that Keycloak nodes would use to discover new peers                    | `nil`                                                   |

Reviewer: ```suggestion
| `serviceDiscovery.protocol`             | Sets the protocol that Keycloak nodes would use to discover new peers                    | `kubernetes.KUBE_PING`                                  |
```

| `containerPorts.http`                 | HTTP port to expose at container level                                                     | `8080`                                                  |
| `containerPorts.https`                | HTTPS port to expose at container level                                                    | `8443`                                                  |
| `podSecurityContext`                  | Keycloak pods' Security Context                                                            | Check `values.yaml` file                                |
| `containerSecurityContext`            | Keycloak containers' Security Context                                                      | Check `values.yaml` file                                |
| `resources.limits`                    | The resources limits for the Keycloak container                                            | `{}`                                                    |
| `resources.requests`                  | The requested resources for the Keycloak container                                         | `{}`                                                    |

> | `lifecycleHooks`                      | LifecycleHooks to set additional configuration at startup.                                 | `{}` (evaluated as a template)                          |

Reviewer: Duplicated parameter. Following `values.yaml` order, I think this is the line that should be deleted.

| `customLivenessProbe`                 | Override default liveness probe                                                            | `nil`                                                   |
| `customReadinessProbe`                | Override default readiness probe                                                           | `nil`                                                   |
| `updateStrategy`                      | Strategy to use to update Pods                                                             | Check `values.yaml` file                                |
| `podAffinityPreset`                   | Pod affinity preset. Ignored if `affinity` is set. Allowed values: `soft` or `hard`        | `""`                                                    |
| `podAntiAffinityPreset`               | Pod anti-affinity preset. Ignored if `affinity` is set. Allowed values: `soft` or `hard`   | `soft`                                                  |
| `nodeAffinityPreset.type`             | Node affinity preset type. Ignored if `affinity` is set. Allowed values: `soft` or `hard`  | `""`                                                    |

> | `nodeAffinityPreset.key`              | Node label key to match Ignored if `affinity` is set.                                      | `""`                                                    |

Reviewer: ```suggestion
| `nodeAffinityPreset.key`              | Node label key to match. Ignored if `affinity` is set.                                     | `""`                                                    |
| `nodeAffinityPreset.values`           | Node label values to match. Ignored if `affinity` is set.                                  | `[]`                                                    |
```

Missing parameter.

### Other parameters

| Parameter                               | Description                                                         | Default                                                 |
|-----------------------------------------|---------------------------------------------------------------------|---------------------------------------------------------|
| `pdb.create`                            | Enable/disable a Pod Disruption Budget creation                     | `false`                                                 |
| `pdb.minAvailable`                      | Minimum number/percentage of pods that should remain scheduled      | `1`                                                     |
| `pdb.maxUnavailable`                    | Maximum number/percentage of pods that may be made unavailable      | `nil`                                                   |
| `autoscaling.enabled`                   | Enable autoscaling for Keycloak                                     | `false`                                                 |

> | `autoscaling.minReplicas`               | Minimum number of Keycloak replicas                                 | `nil`                                                   |
> | `autoscaling.maxReplicas`               | Maximum number of Keycloak replicas                                 | `nil`                                                   |

Reviewer: ```suggestion
| `autoscaling.minReplicas`               | Minimum number of Keycloak replicas                                 | `1`                                                     |
| `autoscaling.maxReplicas`               | Maximum number of Keycloak replicas                                 | `11`                                                    |
```

Per `values.yaml`

|-------------------------------------------|----------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------|
| `metrics.enabled`                         | Enable exposing Keycloak statistics                                                                                  | `false`                                                      |
| `metrics.service.port`                    | Service HTTP managemenet port                                                                                        | `9990`                                                       |
| `metrics.service.annotations`             | Annotations for enabling prometheus to access the metrics endpoints                                                  | `{prometheus.io/scrape: "true", prometheus.io/port: "9990"}` |
| `metrics.serviceMonitor.enabled`          | Create ServiceMonitor Resource for scraping metrics using PrometheusOperator                                         | `false`                                                      |

> | `metrics.serviceMonitor.namespace`        | Namespace which Prometheus is running in                                                                             | `monitoring`                                                 |

Reviewer: ```suggestion
| `metrics.serviceMonitor.namespace`        | Namespace which Prometheus is running in                                                                             | `nil`                                                        |
```

Same as above.

```yaml
sidecars:
  - name: your-image-name
    image: your-image
    imagePullPolicy: Always
    ports:
      - name: portname
       containerPort: 1234
```

Similarly, you can add extra init containers using the `initContainers` parameter.

```yaml
initContainers:
  - name: your-image-name
    image: your-image
    imagePullPolicy: Always
    ports:
      - name: portname
        containerPort: 1234
```

### Initialize a fresh instance

The [Bitnami Keycloak](https://github.com/bitnami/bitnami-docker-keycloak) image allows you to use your custom scripts to initialize a fresh instance. In order to execute the scripts, you can specify custom scripts using the `initdbScripts` parameter as dict.

In addition to this option, you can also set an external ConfigMap with all the initialization scripts. This is done by setting the `initdbScriptsConfigMap` parameter. Note that this will override the previous option.

The allowed extensions is `.sh`.

### Deploying extra resources

> There are cases where you may want to deploy extra objects, such a ConfigMap containing your app's configuration or some extra deployment with a micro service used by your app For covering this case, the chart allows adding the full specification of other objects using the `extraDeploy` parameter.

Reviewer: ```suggestion
There are cases where you may want to deploy extra objects, such a ConfigMap containing your app's configuration or some extra deployment with a micro service used by your app. For covering this case, the chart allows adding the full specification of other objects using the `extraDeploy` parameter.
```

Missing punctuation.
